### PR TITLE
Shepherd/launchbuttons

### PR DESCRIPTION
**Pull request by FIREdog5**

### CODE REVIEW

**Comment by jonathan-j-lee** [comment]
Pull from `master` to get the Dawn version fix. Note that you have a bunch of conflicting files you'll need to fix. @FIREdog5

File: shepherd/Code.py [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+
+def assign_code_solution():
+    '''
+    Generate 16 codes and create a code_solution dictionary
+    '''
+    global codes
+    global solutions
+    global code_solution
+    global code_effect
+    global effect_list
```
Comment (jonathan-j-lee) [inline shepherd/Code.py]: Most of these variables don't need `global`. `global` is needed for changing what these variables point to (e.g. if you wanted `codes = 1`), but is not needed for internal mutations (e.g. appending elements to lists or setting key-value pairs).

File: shepherd/Code.py [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+    for i in range(16):
+        code_solution[codes[i]] = solutions[i]
+    return code_solution
+
+def assign_code_effect():
+    '''
+    Assign each code to a random effect
+    '''
+    for i in range(16):
+        code_effect[codes[i]] = effect_list[random.randint(0, len(effect_list))]
```
Comment (jonathan-j-lee) [inline shepherd/Code.py]: Use `random.choice(effect_list)`.

File: shepherd/Code.py [hunk 3]
```diff
@@ -0,0 +1,45 @@
+import random
+from Utils import *
+
+codes = []
+solutions = []
+code_solution = {}
+code_effect = {}
+effect_list = []
+effect_list = list(EFFECTS.ALL_EFFECTS)
```
Comment (jonathan-j-lee) [inline shepherd/Code.py:8]: Is there a reason this state and code isn't wrapped in a class?

File: shepherd/Code.py [hunk 4]
```diff
@@ -0,0 +1,45 @@
+import random
+from Utils import *
+
+codes = []
+solutions = []
+code_solution = {}
+code_effect = {}
+effect_list = []
```
Comment (jonathan-j-lee) [inline shepherd/Code.py]: Doesn't seem to be used.

**Review by jonathan-j-lee — CHANGES_REQUESTED** [review]
Comments so far.

File: shepherd/Code.py (hunk 2 above)
Comment (FIREdog5) [reply shepherd/Code.py]: Done.

**Review by jonathan-j-lee — APPROVED** [review]

**Pull request merged**